MekHQ/src/mekhq/gui/dialog/nagDialogs/UntreatedPersonnelNagDialog.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (c) 2021 - The MegaMek Team. All Rights Reserved.
+ *
+ * This file is part of MekHQ.
+ *
+ * MekHQ is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * MekHQ is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with MekHQ. If not, see <http://www.gnu.org/licenses/>.
+ */
+package mekhq.gui.dialog.nagDialogs;
+
+import mekhq.MHQConstants;
+import mekhq.MekHQ;
+import mekhq.campaign.Campaign;
+import mekhq.campaign.personnel.Injury;
+import mekhq.campaign.personnel.Person;
+import mekhq.gui.baseComponents.AbstractMHQNagDialog;
+
+import javax.swing.*;
+
+public class UntreatedPersonnelNagDialog extends AbstractMHQNagDialog {
+    private static boolean isUntreatedInjury (Campaign campaign) {
+        for (Person p : campaign.getActivePersonnel()) {
+            if (campaign.getCampaignOptions().isUseAdvancedMedical()) {
+                for (Injury i : p.getInjuries()) {
+                    if (!i.isPermanent()) {
+                        return p.getDoctorId() == null;
```

Comment:
This is wrong. You are going to return true/false on the first person in the loop and then the loop will stop processing. You should have left this how you had it before.  Same for the other case below.

```java
if (!i.isPermanent() && p.getDoctorId() == null) {
  return true;
} 
```